### ISSUE TITLE
Why not a kube-proxy chart?

### ISSUE THREAD

**Issue by clayvan** [aws/eks-charts#154]
EKS is one of the few managed Kubernetes platforms that doesn't provide kube-proxy as a part of the control plane upgrade, as we still have to manage/upgrade kube-proxy during EKS upgrades.

I was surprised to find 0 public helm charts for kube-proxy, and I think it's because it's pretty specific to EKS.

In the meantime I'll create my own. If you think it's a good idea, I'll PR it up to this repo, but was more curious as to why no-one else has one.

**Comment by max-rocket-internet** [comment]
I don't know of a specific reason why there's no chart but my impression is that other than updating the image, there's not much else to for an upgrade.

**Comment by mogren** [comment]
@clayvan the task to manage addons is tracked here https://github.com/aws/containers-roadmap/issues/252

**Comment by TBBle** [comment]
I'm pretty sure kube-proxy doesn't have a chart because everyone is using the Daemonset deployed by kubeadm. I haven't checked my EKS cluster, but I'd be surprised if there was anything different in its deployment except using an ECR-hosted image.

It'd be a pretty simple Helm chart to create, but it's technically not eks-specific, and since [helm/charts ("stable") is being deprecated](https://github.com/helm/charts#status-of-the-project) it's not obvious where it should live. Basically, who wants to maintain it?

**Comment by sc250024** [comment]
I'd totally be for this, coupled with an adoption strategy. I opened the issue (https://github.com/aws/containers-roadmap/issues/923) because of this exact problem.

**Comment by stevehipwell** [comment]
What would be peoples thoughts be on a PR for an EKS specific chart for kube-proxy to be added to this repo? Our use case would be to remove the default config at cluster creation and manage it via Helm thereafter, all in Terraform. The EKS specifics would be around the image selection based on cluster version and region inputs as well as taints etc for Fargate.

**Comment by sc250024** [comment]
> What would be peoples thoughts be on a PR for an EKS specific chart for kube-proxy to be added to this repo? Our use case would be to remove the default config at cluster creation and manage it via Helm thereafter, all in Terraform. The EKS specifics would be around the image selection based on cluster version and region inputs as well as taints etc for Fargate.

Yes! This is especially the case since the above issue (https://github.com/aws/containers-roadmap/issues/923) has apparently been considered, and will be released at some point.

**Comment by stevehipwell** [comment]
@sc250024 I've not seen any meaningful movement on aws/containers-roadmap#923 which is why I'm looking at adding the Helm chart myself. Having to use shell scripts to manipulate _kube-proxy_ and _core-dns_ doesn't go well with infrastructure automation.
